addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/handler/LoxoneMiniserverHandler.java

```diff
@@ -241,32 +239,14 @@ public void channelLinked(ChannelUID channelUID) {
     @Override
     public void initialize() {
         logger.trace("Initializing thing");
-
-        // config.as does not put restrictions on the return time (it is nullable)
-        // but compiler figures out its implementation never returns null and gives a warning
-        // about redundant null check - so we do not check for null here even though this is nullable
-        @Nullable
         LoxoneMiniserverConfig cfg = getConfig().as(LoxoneMiniserverConfig.class);
-        String host = cfg.host;
-        String user = cfg.user;
-        String password = cfg.password;
-        int port = cfg.port;
-        if (user != null && password != null) {
-            try {
-                @SuppressWarnings("null")
-                @NonNull
-                InetAddress ip = InetAddress.getByName(host);
-                LxServer srv = new LxServer(ip, port, user, password);
-                srv.addListener(this);
-                srv.update(cfg.firstConDelay, cfg.keepAlivePeriod, cfg.connectErrDelay, cfg.responseTimeout,
-                        cfg.userErrorDelay, cfg.comErrorDelay, cfg.maxBinMsgSize, cfg.maxTextMsgSize);
-                srv.start();
-                this.server = srv;
-            } catch (UnknownHostException e) {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Unknown host");
+        try {
+            server = createNewServer(cfg);
+            if (server == null) {
```

Comment:
Minor since you are already handling exceptions, maybe you might be better of not returning `null` from `createNewServer()` and throw an exception in that error scenario, then both the OFFLINE markings end up on the same level,